doc/releases/changelog-dev.md

```diff
@@ -2,6 +2,9 @@
 
 <h3>New features</h3>
 
+* Implement detensorization pass for the SCF dialect (for, if, while operations only);
+  [(#1075)](https://github.com/PennyLaneAI/catalyst/pull/1075)
```

Comment:
How about that?

Reply:
> * The `linalg-detensorize` pass detensors internal control flow inside a function, but scalar tensors remain in the entry block. In particular, scalar tensors are constructed and their element are extracted around often used structured control flow operations such as for, if, and while. This PR introduces the `detensorize-scf` pass which removes scalar tensors from the SCF dialect (for, if, while operations). A follow-up `canonicalize` pass is necessary to cancel out and remove `tensor::ExtractOp` and `tensor::FromElementsOp` operations.

The new block is a good explanation for developers of catalyst, but it doesn't explain why you would want to do this to begin. The paragraph could be a good fit for the body of the changelog entry, but lets start the headline with:
 - what has changed? (scalar tensors are eliminated around control flow) 
 - and what does it do for the user? (improves performance)

The main idea is that the changelog is targeted towards our users first, and possibly towards integrators second (who might be interested in internal changes). We are typically not targeting ourselves since we also have PR/commit logs, comments, etc that contain development information.

Reply:
Updated in b4890eb02264d5045ae77ea8848bd781df7c31c0